### PR TITLE
Move some daml-trigger compat tests to sandbox-on-x

### PR DESCRIPTION
**Pull request by sofiafaro-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/bazel_tools/daml_trigger/daml_trigger.bzl [hunk 1]
```diff
@@ -124,6 +124,21 @@ def daml_trigger_test(compiler_version, runner_version):
         compiler_version = version_to_name(compiler_version),
         runner_version = version_to_name(runner_version),
     )
+
+    # 1.16.0 is the first SDK version that uses LF 1.14, which is the earliest version that canton supports
+    use_canton = versions.is_at_least("2.0.0", runner_version) and versions.is_at_least("1.16.0", compiler_version)
+    use_sandbox_on_x = versions.is_at_least("2.0.0", runner_version) and not use_canton
+    if use_sandbox_on_x:
+        server = "@daml-sdk-{version}//:sandbox-on-x".format(version = runner_version)
+        server_args = ["--participant", "participant-id=sandbox,port=6865"]
```
Comment (cocreature) [inline compatibility/bazel_tools/daml_trigger/daml_trigger.bzl:133]: I’m not entirely sure if those tests are marked exclusive. If they are not, we cannot hardcode the port number. Marking them exclusive seems fine as a short term (up to 2.0) fix.

Comment (cocreature) [reply compatibility/bazel_tools/daml_trigger/daml_trigger.bzl:133]: Haven’t checked if the same applies to the daml script tests.

**Review by cocreature — APPROVED** [review]
Nice thank you!

File: compatibility/bazel_tools/daml_trigger/daml_trigger.bzl (hunk 1 above)
Comment (sofiafaro-da) [reply compatibility/bazel_tools/daml_trigger/daml_trigger.bzl:133]: They are tagged exclusive, as are the daml-script tests.

Comment (cocreature) [reply compatibility/bazel_tools/daml_trigger/daml_trigger.bzl:133]: perfect thanks for double checking

**Pull request merged**